File: admin/entity/database/review.go
package database

import "time"

type Review struct {
	Id        string
	Code      string    `firestore:"code"`
	Comment   string    `firestore:"comment"`
	CreatedAt time.Time `firestore:"createdAt,serverTimestamp"`
	Deleted   bool      `firestore:"deleted"`
	Images    []string  `firestore:"images"`
	Rate      int       `firestore:"rate"`
	Uid       string    `firestore:"uid"`
	UpdatedAt time.Time `firestore:"updatedAt,serverTimestamp"`
}

//func (a *Review) Data() map[string]interface{} {
//	return map[string]interface{}{
//		"code":      a.Code,
//		"comment":   a.Comment,
//		"createdAt": a.CreatedAt,
//		"deleted":   a.Deleted,
//		"images":    a.Images,
//		"rate":      a.Rate,
//		"uid":       a.Uid,
//		"updatedAt": a.UpdatedAt,
//	}
//}

File: ios/Reviewers/Entity/Review.swift
import Foundation

struct Review: Identifiable, Hashable {
    let id: String
    let uid: String
    let deleted: Bool
    let code: String
    let codeType: CodeType
    let comment: String
    let images: [String]
    let rate: Int
    let createdAt: Date
    let updatedAt: Date
    let storageEndpoint: String

    var createdAtString: String {
        let formatter = DateFormatter()
        formatter.dateStyle = .short
        formatter.timeStyle = .medium
        return formatter.string(from: createdAt)
    }

    var imageUrls: [URL] {
        return images.compactMap { URL(string: "\(storageEndpoint)/image/user/\(uid)/\($0)") }
    }

    var profileImageUrlString: String {
        return "\(storageEndpoint)/image/user/\(uid)/profile.png"
    }
}
